### PR TITLE
Remove the extra part from hover message

### PR DESCRIPTION
**Pull request by xphoniex**
Should fix https://github.com/zigtools/zls/issues/1376.

Please let me know if there should be any cases where we have to show `resolved_type_str` too.

### CODE REVIEW

**Comment by Techatrix** [comment]
The resolved_type generally should be shown. Hovering over functions should be the exception where it's unnecessary. The hover when markdown is enabled should also be updated.

**Comment by xphoniex** [comment]
Can you give me a few examples (in code), what should show and what shouldn't, so I can test locally before pushing the fix?

**Comment by xphoniex** [comment]
Check https://github.com/zigtools/zls/pull/1377/commits/a77cac1ba3ba8cd66d471285f835c8432a33ec05 :

Screenshot from 2023-08-02 10-15-46

Screenshot from 2023-08-02 10-16-25

File: src/features/hover.zig [hunk 1]
```diff
@@ -122,7 +129,11 @@ pub fn hoverSymbol(
             try writer.print("[{s}]({s}#L{d})", .{ ref.str, ref.handle.uri, line });
         }
     } else {
-        try writer.print("{s} ({s})", .{ def_str, resolved_type_str });
+        if (is_fn) {
+            try writer.print("{s}", .{def_str});
+        } else {
+            try writer.print("{s} ({s})", .{ def_str, resolved_type_str });
```
Comment (Techatrix) [inline src/features/hover.zig:135]: I know this is unrelated to your fix but while were modifying hover. I think there should be a line break between the definition string and resolved type just like in the markdown version.

Comment (xphoniex) [reply src/features/hover.zig]: [diff](https://github.com/zigtools/zls/compare/a77cac1ba3ba8cd66d471285f835c8432a33ec05..5d84a617b9bd714735bc45f68098ef210ae58edb)

**Pull request merged**